src/functional/ffront/func_to_foast.py

```diff
@@ -42,6 +42,34 @@ class FieldOperatorSyntaxError(DialectSyntaxError):
     dialect_name = "Field Operator"
 
 
+class UnrollPowerOp(eve.NodeTranslator):
+    def visit_BinOp(self, node: foast.BinOp) -> foast.BinOp:
+        if node.op == foast.BinaryOperator.POW:
+            if not type(node.right) is foast.Constant:
+                raise ValueError(
```

Comment:
We follow the strict guideline of all errors occurring in the frontend to give information about their location in the original source code. See [here](https://github.com/GridTools/gt4py/blob/67d97b2e9cc4dbf4cdf52973d8a81b33927f1041/src/functional/ffront/foast_passes/type_deduction.py#L393) for a template on how to define such an exception. Given the low length of the pass and its probable fate, adding a new exception feels not very elegant, but I believe this is best way to go for now.